interp/constrintern.ml

```diff
@@ -2471,6 +2471,15 @@ let intern_constr_pattern env sigma ?(as_type=false) ?(ltacvars=empty_ltac_sign)
             ~pattern_mode:true ~ltacvars env sigma c in
   pattern_of_glob_constr c
 
+let interp_constr_pattern env sigma ?(expected_type=WithoutTypeConstraint) c =
+  let kind_for_intern = match expected_type with OfType _ -> WithoutTypeConstraint | _ -> expected_type in
```

Comment:
This line is very strange.

Reply:
> This line is very strange.

I copied it from `tacinterp.ml` but it is indeed strange and actually useless (same job done by `allowed_binder_kind_of_type_kind` already).

Changed.